File: MultipleBaseCase/recurplot2.m
function y=recurplot2(x,n)

y=[];
basecase=(1:n);

recur(x);

%Remove adjacent duplicates, there are some issues in running recursive functions in a
%forloop.
y(diff(y)==0)=[];

    function recur(x)
        if ismember(x,basecase)
            y(end+1)=x;
            return
        end
        
        for i=1:n
            y(end+1)=x;
            recur(x-i);
            y(end+1)=x;
        end
        %% Can also set depth of recursion this way.
        
        %         y(end+1)=x;
        %         recur(x-1)
        %         y(end+1)=x;
        %         recur(x-2)
        %         y(end+1)=x;
        %         recur(x-3);
        %         y(end+1)=x;
    end

%% Plotting
plot(y, '-o')
ylabel('X value')
xlabel('Stack Step')
title(strcat('Stack:', num2str(x), '   Base Cases:', num2str(n)))
end

File: MultipleBaseCase/recurplot2.py
import matplotlib.pyplot as plt

def recurplot2(x,n):
	y=[]
	print(x,n)
	basecase = range(1,n+1)

	def recur(x):
		if x in basecase:
			y.append(x)
			return

		for i in range(1,n+1):
			y.append(x)
			recur(x-i)
			y.append(x)

	recur(x)
	
	#still have to remove adjacent duplicates
	y_nodupe = [x for i, x in enumerate(y) if i==0 or x!=y[i-1]]

	#Plotting
	plt.plot(y_nodupe, '-o')
	plt.xlabel('Stack Step')
	plt.ylabel('X Value')
	plt.show()

	return y, y_nodupe